### PR TITLE
Test that LVM device is encrypted.

### PR DESCRIPTION
**Pull request by mulkieran**
Signed-off-by: mulhern <amulhern@redhat.com>

### CODE REVIEW

**Comment by mulkieran** [comment]
I thought of setting flags -e -o pipefail using %post --interpreter option, but it seems like

```
%post --interpreter="/usr/bin/bash -e -o pipefail"
```
but I think that is doomed to fail because entire string "/usr/bin/bash -e -o pipefail" is interpreted as the command to be passed to subprocess.

File: tests/kickstart_tests/encrypt-device.ks [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+lang en_US.UTF-8
+timezone America/New_York --utc
+rootpw testcase
+shutdown
+
+%packages --default
+%end
+
+%post
+if [ -z "$(blkid -t TYPE="crypto_LUKS")" ]; then
```
Comment (dwlehman) [inline tests/kickstart_tests/encrypt-device.ks]: You could do something a bit more demanding along these lines:

```sh
by_path=$(blkid -odevice -t LABEL=root | grep -q '^/dev/mapper/luks-')
by_type=$(blkid -ovalue -t LABEL=root -sTYPE | grep crypto_LUKS)
if [ -z "$by_path" -a -z "$by_type" ]; then
    echo "root LV is not encrypted" > /root/RESULT
fi
```

**Comment by bcl** [comment]
> but I think that is doomed to fail because entire string "/usr/bin/bash -e -o pipefail" is interpreted as the command to be passed to subprocess.

You can also use 'set -e' and 'set -o pipefail' inside the %post script which should work.

**Comment by mulkieran** [comment]
Unfortunately, the label is no help in locating the luks device directly.

blkid -t LABEL=root gives me two devices. They have device ```/dev/block/253:6``` and ```/dev/dm-6``` respectively, and both have type ```ext4```. They are the LUKS device, but the data is unsatisfactory. Connecting the root lv to the encrypted device seems quite tricky, except in the nice visual representation that lsblk has, which is unhandy to parse.

**Comment by dwlehman** [comment]
My mistake -- for the second test you'd do this:

`by_type=$(blkid -ovalue  -sTYPE /dev/mapper/vg01-root_lv | grep crypto_LUKS)`

or you could test it for equality since it can only output one line:

```sh
by_path=$(blkid -odevice -t LABEL=root | grep -q '^/dev/mapper/luks-')
root_lv_type="$(blkid -ovalue  -sTYPE /dev/mapper/vg01-root_lv)"
if [ -z "$by_path" -a "$by_type" != "crypto_LUKS" ]; then
    echo "root LV is not encrypted" > /root/RESULT
fi

**Comment by mulkieran** [comment]
I couldn't get any observable benefit from the ```set -e``` and ```-o pipefail``` commands. It seems like they don't help with enough kinds of mistakes to do much more than provide a false sense of security.
So I skipped them.

I'm not sure how the ```by_path``` variable could every get a value, so I skipped using that do.

**Comment by dwlehman** [comment]
See the updated comment text -- I just accidentally left "by_path" instead of replacing it with "root_lv_type".

**Comment by mulkieran** [comment]
ok, good, then I think this is ackable.

**Comment by dwlehman** [comment]
Oh, I was too hasty. All you have to do is omit the `-q` from the grep command. Either of those tests should be sufficient, I guess.

**Comment by mulkieran** [comment]
OK! I'll stay, then. There is no "/dev/mapper/luks-" name device results connected with LABEL=root that I have observed.

**Comment by vpodzime** [comment]
Looks good to me.

**Pull request merged**